### PR TITLE
Made Java installation as connection features

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
This PR takes the Java discovery part of https://github.com/codefori/vscode-ibmi/pull/2195.
The different possible Java installation locations are made into connection features.

These features are now used to get Java Home path when configuring the debug service.

### How to test this PR
1. Run the Debug test case
2. Generate a Debug Service certificate
3. Check Java Home in the Debug Service configuration file

### Checklist
* [x] have tested my change

### CODE REVIEW

**Comment by worksofliam** [comment]
Beautiful code change. Will get it tested ASAP.

File: src/api/IBMi.ts [hunk 1]
```diff
@@ -587,6 +591,16 @@ export default class IBMi {
                 console.log(e);
               }
             }
+
+            //Specific Java installations check
+            progress.report({
+              message: `Checking installed components on host IBM i: Java`
+            });
+            const javaCheck = async (root: string) => await this.content.testStreamFile(`${root}/bin/java`, 'x') ? `${root}` : undefined;
+            this.remoteFeatures.jdk80 = await javaCheck(`/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit`);
```
Comment (worksofliam) [inline src/api/IBMi.ts:600]: Just being specific. Should this be `jdk08` or is `jdk80` ok?

File: src/api/IBMi.ts [hunk 2]
```diff
@@ -587,6 +591,16 @@ export default class IBMi {
                 console.log(e);
               }
             }
+
+            //Specific Java installations check
+            progress.report({
+              message: `Checking installed components on host IBM i: Java`
+            });
+            const javaCheck = async (root: string) => await this.content.testStreamFile(`${root}/bin/java`, 'x') ? `${root}` : undefined;
```
[the remark this answers is not in the document]
Comment (worksofliam) [reply src/api/IBMi.ts:599]: Just a note: I had to connect and reload for this to be hit.

**Review by worksofliam — CHANGES_REQUESTED** [review]
I think my only concern is the finding of JDK version without reloading. Other than that, very nice change!

File: src/api/IBMi.ts (hunk 1 above)
Comment (sebjulliand) [reply src/api/IBMi.ts:600]: `jdk80` it is.
image

**Comment by sebjulliand** [comment]
> I think my only concern is the finding of JDK version without reloading. Other than that, very nice change!

@worksofliam Can you try again? I connected to an existing system created and connected before the PR, and the reloading occurred as expected.

The existing test was accurate:
image

Or am I missing something obvious? 😅

**Review by worksofliam — APPROVED** [review]
@sebjulliand Tested again and it worked as expected. I connected to an existing system and it grabbed the JDK correctly. Thanks!

Merge away!

**Pull request merged**